pkg/apis/config/v1alpha2/meshrootcertificate.go

```diff
@@ -112,15 +112,10 @@ type TresorCASpec struct {
 }
 
 // MeshRootCertificateStatus defines the status of the MeshRootCertificate resource
-type MeshRootCertificateStatus struct {
+type ProviderState struct {
 	// State specifies the state of the root certificate rotation
+	// All states are specified in constants.go
 	State string `json:"state"`
```

Comment:
More files need to be updated to support this change. For example `config_mesh_root_certificate.yaml` and any other references/logic pertaining to the current status options.